.gitlab-ci.yml

```diff
@@ -265,6 +265,32 @@ pkg:nix:
     paths:
       - nix-build-coq.drv-0/*/test-suite/logs
 
+build:macOS:
+  <<: *build-template
+  tags:
+    - macOS
+  before_script:
+    - brew update
+    - brew unlink python
+    - brew install opam gnu-time gtk+ expat gtksourceview gdk-pixbuf
+    - brew unlink python@2
+    - brew link python3
+    - pip3 install macpack
+    - opam init -a -y -j 2 --compiler=4.02.3 default https://opam.ocaml.org
+    - eval $(opam config env)
+    - opam update
+    - opam config list
+    - opam list
+    - opam install -j 2 -y camlp5.6.17 ocamlfind num lablgtk.2.18.6 lablgtk-extras.1.6
+    - rm -rf ~/.opam/log/
+    - opam list
+  after_script:
+    - dev/build/osx/make-macos-dmg.sh
```

Comment:
Remove this `after_script`. This caused an infinite loop of @coqbot retrying the failed job. The job of `after_script` is not how you use it anyways.

Reply:
What should I put instead?

Reply:
You should put this in the main script. `after_script` is something that is executed *even if the build failed*.

Reply:
> You should put this in the main script.

Ok, but how do I do that?

Reply:
I'm not sure that reusing build-template is the right thing to do here.

Reply:
So just overwrite the default `script`.

Reply:
>This caused an infinite loop of @coqbot

We should have a retry limit to avoid that/

Reply:
I have looked into this and I haven't found any way of enforcing that. But this should be a really rare occurrence I think (can only happen if someone is overriding `after_script` in a PR and can be stopped by cancelling the build).